pkg/controller/controller.go

```diff
@@ -907,6 +907,12 @@ func (c *M3DBController) updateStatefulSetPods(
 			if err := c.kubeClient.CoreV1().
 				Pods(pod.Namespace).
 				Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
+				if kerrors.IsNotFound(err) {
+					// This pod was already deleted in a previous iteration of the reconile loop. This can happen
```

Comment:
This was a real issue in the reconcile loop, due to the indexer being stale.

1. Delete Pod
2. Loop again and list pods. Pod still in index.
3. Indexer updated.
4. Delete Pod again but no longer in Indexer.